xvec/index.py

```diff
@@ -0,0 +1,193 @@
+                    "a single geometry as input label."
+                )
+
+        if isinstance(label, DataArray):
+            label_array = label._variable._data
+        elif isinstance(label, Variable):
+            label_array = label._data
+        elif isinstance(label, shapely.Geometry):
+            label_array = np.array([label])
+        else:
+            label_array = np.array(label)
+
+        # check for possible CRS of geometry labels
+        # (by default assume same CRS than the index)
+        if hasattr(label_array, "crs") and label_array.crs != self.crs:
+            raise ValueError("conflicting CRS for input geometries")
+
+        assert np.all(shapely.is_geometry(label_array))
+
+        if method == "nearest":
+            indices = self.sindex.nearest(label_array)
+        else:
+            indices = self.sindex.query(label, predicate=method, distance=tolerance)
+
+        # attach dimension names and/or coordinates to positional indexer
+        if isinstance(label, Variable):
+            indices = Variable(label.dims, indices)
+        elif isinstance(label, DataArray):
+            indices = DataArray(indices, coords=label._coords, dims=label.dims)
+
+        return IndexSelResult({self._index.dim: indices})
+
+    def sel(
+        self, labels: dict[Any, Any], method=None, tolerance=None
+    ) -> IndexSelResult:
+        if method is None:
+            return self._index.sel(labels)
+        else:
+            # We reuse here `method` and `tolerance` options of
+            # `xarray.indexes.PandasIndex` as `predicate` and `distance`
+            # options when `labels` is a single geometry.
+            # Xarray currently doesn't support custom options
+            # (see https://github.com/pydata/xarray/issues/7099)
+            return self._sel_sindex(labels, method, tolerance)
+
+    def equals(self, other: Index) -> bool:
+        if not isinstance(other, GeoVectorIndex):
+            return False
+        if other.crs != self.crs:
+            return False
+        return self._index.equals(other._index)
+
+    def join(
+        self: GeoVectorIndex, other: GeoVectorIndex, how: str = "inner"
+    ) -> GeoVectorIndex:
+        index = self._index.join(other._index, how=how)
+        return type(self)(index, self.crs)
+
+    def reindex_like(
+        self, other: GeoVectorIndex, method=None, tolerance=None
+    ) -> dict[Hashable, Any]:
+        return self._index.reindex_like(
+            other._index, method=method, tolerance=tolerance
+        )
+
+    def roll(self, shifts: Mapping[Any, int]) -> GeoVectorIndex:
+        index = self._index.roll(shifts)
+        return type(self)(index, self.crs)
+
+    def rename(self, name_dict, dims_dict):
+        index = self._index.rename(name_dict, dims_dict)
+        return type(self)(index, self.crs)
+
+    def _repr_inline_(self, max_width):
+        return f"{self.__class__.__name__}(crs={self.crs.to_string()})"
```

Comment:
```suggestion
        return f"{self.__class__.__name__} (crs={self.crs.to_string()})"
```
I'd add a space so it doesn't look like a function call.